### DISCUSSION TITLE
External reaction But Time-Varying Force with Changing Magnitude and Direction

### DISCUSSION THREAD

**Discussion by YCJ0815** [JSBSim-Team/jsbsim#1059] [Q&A]
Hello everyone,

I am currently working on constructing a tether wing model that involves considering a force which varies over time not only in magnitude but also in direction. However, the external reaction that defined in XML file can not be time-varying. I would like to inquire about the best approach to incorporate such dynamic forces into my simulation model.

I am particularly interested in finding effective solutions to this problem and would appreciate any suggestions, experiences, or insights you may have.

Thank you in advance for your assistance!

**Comment by andgi** [comment]
I think you can change both magnitude and also direction already.

For direction you can also split it into 3 90 degree angle forces - that has worked from the beginning of external forces.

Have a look at the hydrodynamics.xml system (https://github.com/JSBSim-Team/jsbsim/blob/master/systems/hydrodynamics.xml) for an example. It is used in the aircraft Short_S23 that is in the aircraft directory.

**Comment by seanmcleod** [comment]
Looks like you can change both the direction and magnitude at runtime, so you can make them both time varying. If you use a `function` for the force magnitude you can make use of properties in the `function` which means you can also make it time varying by varying the value of the properties used.

https://github.com/JSBSim-Team/jsbsim/blob/b9bfc2350e0e00e53246b65979cc18df2922a88d/src/models/FGExternalForce.h#L176-L183

https://github.com/JSBSim-Team/jsbsim/blob/b9bfc2350e0e00e53246b65979cc18df2922a88d/src/models/FGExternalForce.h#L196-L201

**Comment by YCJ0815** [comment]
Thank you very much for taking the time to answer my questions, it's been very helpful.
